api/BUILD

```diff
@@ -3,5 +3,6 @@ package(default_visibility = ["//visibility:public"])
 cc_library(
     name = "api",
     hdrs = glob(["include/**/*.h"]),
+    srcs = glob(["src/**/*.cc"]),
```

Comment:
How can we do this in a way that keeps the api header-only?

Reply:
@tigrannajaryan posted this approach that should work: https://gitter.im/open-telemetry/opentelemetry-cpp?at=5d9bde66874aeb2d2308ccda